api/v1/helpers/verbs/doPost.js

```diff
@@ -45,9 +46,17 @@ function doPost(req, res, next, props) {
       featureToggles.isFeatureEnabled('enforceWritePermission'))
       .then((user) => redisModelSample.postSample(req.swagger.params, user));
   } else {
-    postPromise = featureToggles.isFeatureEnabled('enforceWritePermission') &&
-      props.modelName === 'Sample' ? u.createSample(req, props) :
-      props.model.create(toPost);
+    if (featureToggles.isFeatureEnabled('enforceWritePermission') &&
+      props.modelName === 'Sample') {
+      postPromise = u.createSample(req, props);
+    } else {
+      postPromise = authUtils.getUser(req)
+      .then((user) => {
+        toPost.createdBy = user.id;
```

Comment:
will throw error when no user found -- should just leave createdBy empty when no user record found